### PR TITLE
feat: cache instantiated plugins and resolve realm-specific properties

### PR DESCRIPTION
**Pull request by volodymyr-rutskyi**
## What/Why/How?

* Added a simple cache for instantiated plugins. This makes sure that the init logic in instantiated plugins is only executed once. This is relevant for Realm.
* Added resolving of properties like `processContent`, `afterRoutesCreated`, and others that are used by Realm. This enables Realm to use `openapi-core` for plugin resolving and thus keep this logic in one place.

## Reference

Internal request: https://github.com/Redocly/redocly/issues/10463

## Check yourself

- [ ] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [ ] All new/updated code is covered with tests
- [ ] New package installed? - Tested in different environments (browser/node)

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: .changeset/tame-laws-film.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/openapi-core": minor
+"@redocly/cli": minor
+---
+
+Added a cache for resolved plugins that makes sure that plugins are only instantiated once during a single execution.
```
Comment (JLekawa) [inline .changeset/tame-laws-film.md:6]:
```suggestion
Added a cache for resolved plugins to ensure that plugins are only instantiated once during a single execution.
```

File: .changeset/flat-days-train.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@redocly/openapi-core": minor
+---
+
+Added resolving of plugin properties specific to the Reunite-hosted product family.
```
Comment (JLekawa) [inline .changeset/flat-days-train.md:5]:
```suggestion
Added a mechanism that resolves plugin properties specific to the Reunite-hosted product family.
```

File: packages/core/src/config/config-resolvers.ts [hunk 3]
```diff
@@ -313,7 +330,10 @@ export async function resolvePlugins(
         plugin.assertions = pluginModule.assertions;
       }
 
-      return plugin;
+      return {
+        ...pluginModule,
```
[the remark this answers is not in the document]
Comment (volodymyr-rutskyi) [reply packages/core/src/config/config-resolvers.ts:334]: They're already there

**Review by tatomyr — APPROVED** [review]
LGTM

**Pull request merged**